AdyenActions/Components/SDK/TwintSDKActionComponent.swift

```diff
@@ -1,130 +1,130 @@
+        public init(context: AdyenContext,
+                    configuration: Configuration = .init()) {
+            self.context = context
+            self.configuration = configuration
+        }
+
+        /// Handles TwintSDK action.
+        ///
+        /// - Parameter action: The Twint SDK action object.
+        public func handle(_ action: TwintSDKAction) {
+            AdyenAssertion.assert(message: "presentationDelegate is nil", condition: presentationDelegate == nil)
+            Twint.fetchInstalledAppConfigurations { [weak self] installedApps in
+                if installedApps?.count == 0 {
+                    self?.handleShowError("No or an outdated version of TWINT is installed on this device. Please update or install the TWINT app.")
+                } else if installedApps?.count == 1 {
+                    guard let installedApps else {
+                        return
+                    }
+                    self?.invokeTwintAppWithCode(app: installedApps[0], code: action.sdkData.token)
+                } else {
+                    self?.presentAppChooser(installedApps: installedApps ?? [], code: action.sdkData.token)
                 }
-                self?.invokeTwintAppWithCode(app: installedApps[0], code: action.sdkData.token)
-            } else {
-                self?.presentAppChooser(installedApps: installedApps ?? [], code: action.sdkData.token)
             }
         }
-    }
 
-    private func invokeTwintAppWithCode(app: TWAppConfiguration, code: String) {
+        private func invokeTwintAppWithCode(app: TWAppConfiguration, code: String) {
 
-        let error = Twint.pay(withCode: code, appConfiguration: app, callback: callback)
-        if let error = error {
-            AdyenAssertion.assertionFailure(message: error.localizedDescription)
+            let error = Twint.pay(withCode: code, appConfiguration: app, callback: callback)
+            if let error {
+                AdyenAssertion.assertionFailure(message: error.localizedDescription)
+            }
         }
-    }
 
-    private func presentAppChooser(installedApps: [TWAppConfiguration], code: String) {
-        let appChooserViewController = Twint.controller(for: installedApps) { selectedApp in
-            self.invokeTwintAppWithCode(app: selectedApp ?? installedApps[0],
-                                        code: code)
-        } cancelHandler: {
-            self.delegate?.didFail(with: ComponentError.cancelled, from: self)
-        }
-        if let viewcontroller = appChooserViewController, let delegate = presentationDelegate {
-            present(viewcontroller, presentationDelegate: delegate)
+        private func presentAppChooser(installedApps: [TWAppConfiguration], code: String) {
+            let appChooserViewController = Twint.controller(for: installedApps) { selectedApp in
+                self.invokeTwintAppWithCode(app: selectedApp ?? installedApps[0],
```

Comment:
are these closures non escaping? if not, let's add weak

Reply:
I added it 👍